exporters/stdout/stdoutlog/exporter.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package stdoutlog // import "go.opentelemetry.io/otel/exporters/stdout/stdoutlog"
+
+import (
+	"context"
+	"encoding/json"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel/sdk/log"
+)
+
+var zeroTime time.Time
+
+var _ log.Exporter = &Exporter{}
+
+// Exporter writes JSON-encoded log records to an [io.Writer] ([os.Stdout] by default).
+// The writer is os.Stdout by default.
+type Exporter struct {
+	encoder    *json.Encoder
+	timestamps bool
+
+	stopped atomic.Bool
+}
+
+// New creates an Exporter with the passed options.
+func New(options ...Option) (*Exporter, error) {
+	cfg := newConfig(options)
+
+	enc := json.NewEncoder(cfg.Writer)
+	if cfg.PrettyPrint {
+		enc.SetIndent("", "\t")
+	}
+
+	return &Exporter{
+		encoder:    enc,
+		timestamps: cfg.Timestamps,
+	}, nil
+}
+
+// Export exports log records to writer.
+func (e *Exporter) Export(ctx context.Context, records []log.Record) error {
+	if e.stopped.Load() {
+		return nil
+	}
+
+	// Prevent panic if encoder is nil.
+	if e.encoder == nil {
+		e.encoder = json.NewEncoder(defaultWriter)
+	}
```

Comment:
If the exporter would have to be concurrent safe then this can lead to a data race.

However, I think that the exporters do not need to be concurrent safe and `stopped` can be a simple `bool`.

See: https://github.com/open-telemetry/opentelemetry-go/pull/5163#discussion_r1555874107

Reply:
I was wrong. https://github.com/open-telemetry/opentelemetry-go/issues/5178

Reply:
Can you also add a concurrent-safe test?

Reply:
Idea:
1.You can add a comment `Exporter must be created with [New].` 
2. Implement the exporter in a way that an exporter created with `&Exporter{}` (or `new(Expoter)`) will not panic - it can even act as a shut down exporter.

Reply:
> so that the zero value would be like for a stopped one

you mean act like a noop-exporter?

Reply:
Or 3. Not freeing the encoder resources right after the `Shutdown`.

I slightly prefer having option 1, as leaving a noop-exporter to users does not have much meaning of it.

Reply:
I mean we can do both 1. and 2.

Reply:
I also removed the following line to avoid a data race. We can use a mutex to clean the encoder after the shutdown and prevent race conditions, but I am not sure introducing a lock here is worth the effort.

```go
		// Free the encoder resources.
		e.encoder = nil
```

@MrAlias You might need to take a look on

Reply:
You can use a https://pkg.go.dev/sync/atomic#Pointer

Reply:
Though it is kind of an overkill, based on the description of the exporter interface, I think we should still need to free the encoder.

https://github.com/open-telemetry/opentelemetry-go/blob/b9752eb5dc193354a5b19453a6dea3e149a3c056/sdk/log/exporter.go#L38-L39